frontend/app/views/spree/taxons/_subcategories.html.erb

```diff
@@ -1,7 +1,6 @@
 <% cache([base_cache_key, taxon, 'subcategories', permitted_product_params]) do %>
   <% [taxon.parent, taxon].compact.each do |current_taxon| %>
-    <% child_taxons = [current_taxon, *current_taxon.children] %>
-
+    <% child_taxons = [current_taxon, *current_taxon.children.select{|taxon| !taxon.hide_from_nav}] %>
```

Comment:
it might be better to use .where statement here:
children.where(hide_from_nav: false)